src/app/clusters/descriptor/descriptor.cpp

```diff
@@ -283,7 +274,7 @@ CHIP_ERROR DescriptorAttrAccess::Read(const ConcreteReadAttributePath & aPath, A
         return ReadTagListAttribute(aPath.mEndpointId, aEncoder);
     }
     case ClusterRevision::Id: {
-        return ReadClusterRevision(aPath.mEndpointId, aEncoder);
+        return aEncoder.Encode(kRevision);
```

Comment:
The interesting part is that the metadata header auto-updates with the XML whereas the implementation here does not.  So it's very easy for us to start claiming a new cluster revision without actually implementing it correctly....

I feel like we should in fact static_assert that kRevision is the one this file implements, and then manually change that assert when the revision changes _and we implement the new bits_.

Reply:
This might be a general problem for our cluster implementations, by the way....

Reply:
The kRevision *is*  the one from the Metadata.h...

Reply:
For fully code-backed clusters, like Basic Info and Descriptor, there is NO DIFFERENCE :)

Reply:
How is there no difference?

Concrete example: say the spec in revision X+1 adds a mandatory attribute.  In our implementation, we don't implement that attribute yet.  We should keep reporting revision X until we implement it, no?